project/code/keychain/store.py

```diff
@@ -0,0 +1,62 @@
+"""
+KeyChain key-value store (stub).
+
+NB: Feel free to extend or modify.
+"""
+
+from keychain import Blockchain
+from keychain import Transaction
+
+
+class Callback:
+    def __init__(self, transaction, chain):
+        self._transaction = transaction
+        self._chain = chain
+
+    def wait(self):
+        """Wait until the transaction appears in the blockchain."""
+        raise NotImplementedError
+
+    def completed(self):
+        """Polls the blockchain to check if the data is available."""
+        raise NotImplementedError
+
+
+class Storage:
+    def __init__(self, bootstrap, miner, difficulty):
+        """Allocate the backend storage of the high level API, i.e.,
+        your blockchain. Depending whether or not the miner flag has
+        been specified, you should allocate the mining process.
+        """
+        self._blockchain = Blockchain()
+        raise NotImplementedError
+
+    def put(self, key, value, block=True):
```

Comment:
Provided we ask them to run the blockchain behind a rest service, shouldn't we exemplify here how to make a (REST) request instead? I foresee confusion otherwise.

Reply:
I would abstract it away within the Blockchain class, makes the API a bit cleaner imho, e.g.,

```python
def put(key, value):
    transaction = Transaction(key, value)
    self._blockchain.do_transaction(transaction)
    # `do_transaction` does the broadcast (and the appending to the block if the mining process has been specified).
```

Reply:
Alright, I rephrased the instructions to simply say that network communication should be implemented through a rest api.